### ISSUE TITLE
MessageBuilder should be able to add/set custom fields for message

### ISSUE THREAD

**Issue by cuonghuunguyen** [RocketChat/Rocket.Chat.Apps-engine#309]
In App-engine's IMessage interface, we have the field for storing custom fields but in the MessageBuilder, there is no way to add a custom field to the Message. Could you provide a method to add custom fields to the message?

**Comment by matheusbsilva137** [comment]
Hi, @meomay503 !

There is one method (addCustomField) already implemented and available in the `messageExtender` object, which can just be used after the message is sent.
The following repository may guide you through the usage of the `messageExtender`: https://github.com/RocketChat/Apps.RocketChat.Tester/tree/recipes/extending-messages

In brief, the process of sending a message returns an ID, which can be used in order to obtain the messageExtender that allows the addition of custom fields and attachments. The described process can be done as follows:

```
public async executor(context: SlashCommandContext, read: IRead, modify: IModify, http: IHttp, persis: IPersistence): Promise<void> {
    const messageId = await this.sendMessage(context, modify, 'Sending a message!');
    const messageExtender = await this.getMessageExtender(context, modify, messageId);
    const value = 1;
    const img = new ImageAttachment('https://open.rocket.chat/images/logo/logo.svg');
        
    messageExtender.addCustomField('key', value);
    messageExtender.addAttachment(img);

    await modify.getExtender().finish(messageExtender);
}
```

The complete file for the piece of code above can be found in: https://github.com/RocketChat/Apps.RocketChat.Tester/blob/recipes/extending-messages/slashcommands/ExtendSlashcommand.ts

**Comment by meomay503** [comment]
Hi @matheusbsilva137, thank you for the hint. Well, your approach works and I tried it before for my use case. The problem is the latency between the moment the message is sent and the time it gets updated. 
It seems possible to create a message with custom fields and attachments by API or DDP. The current Apps Engine API also allows us to send messages with custom fields by IMessageBuilder when using with tsignore. I think the only missing thing here is the ts definition for that :D. Please correct me if you guys have any intention to disable that feature ^^

**Comment by matheusbsilva137** [comment]
Hi, @meomay503!

We have performed some tests on the latency time using the `messageExtender`, and the results we obtained show that the elapsed time between sending a message and editing it depends on the quantity of new data attached to the message.
However, if the latency time is still a problem for you, I suggest you to keep track of our new **Pull Request #363**. The `addCustomField` method may be available soon for you using only the `messageBuilder` object (which does not perform message extensions; hence, there is no latency time). 😄

About the **API**, it is currently possible to send messages with attachments with no editions, but it still does not support custom fields within messages.
Finally, I must say that we highly suggest you to apply the implementation described in the Pull Request #363 rather than using `tsignore`.
